### DISCUSSION TITLE
Request a method of shell.Command to input sth by stdin.

### DISCUSSION THREAD

**Discussion by lity** [tauri-apps/tauri#2984] [Q&A] [answered]
We can run shell program and get its stdout/stderr, but how to input something to the program by stdin ?

![1](https://user-images.githubusercontent.com/3319045/143891039-350d2860-3e4b-47ef-b3c0-469e0a21832e.jpg)

**Comment by FabianLars** [comment, marked as answer]
you can use the `write()` method on the `Child` object returned by `spawn()`: https://tauri.studio/en/docs/api/js/classes/shell.Child#write

**Reply by lity** [reply]
Yes, that's it. Thx.

**Reply by mauriver21** [reply]
I've implemented the .write() method to stdin a .sh file, but still not workable. Tauri is not able to pass the message to a shell script that is asking for input.
